password_validator.rb

```diff
@@ -3,5 +3,17 @@
 # - include at least one of these characters: !, @, $, %, &
 
 def valid?(password)
-  
+  if password.length >= 8 && password.include?('!')
+    return true
+  elsif password.length >= 8 && password.include?('@')
+    return true
+  elsif password.length >= 8 && password.include?('$')
+    return true
+  elsif password.length >= 8 && password.include?('%')
+    return true
+  elsif password.length >= 8 && password.include?('&')
```

Comment:
These if statements are great but there are ways to refactor them. One example is the method I used and the other is shown on the refactoring page which is next in the tutorials. They'll save loads of time once you get the hang of them

Reply:
Thanks - its definitely something I'm going to read up on and look to implement more.

Reply:
Perhaps the following resource will help:
https://www.rubyguides.com/2015/06/ruby-regex/

Reply:
brilliant - thank you